File: aeroporto.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

int main()
{
    int passageiros;
    char rg[30],Passagem[30],dataRg[30],dataPass[30],cadeiraPass[30];
    printf("Bem vindo ao aeroporto\n");

    printf("Quantos passageiros tem no voo?");
    scanf("%d", &passageiros);

    while (getchar() != '\n');

    for(int i = 0; i < passageiros; i++){
        printf("Passageiro %d:\n", i+1);
        printf("Voce possui RG?[se sim digite: RG/se nao digite: Nao possui]");
        fgets(rg, sizeof(rg), stdin);
        rg[strcspn(rg, "\n")] = '\0';
        if (strcmp(rg, "Nao possui") == 0){
            printf("A saida e nessa direcao\n");
            continue;
        }
        else if (strcmp(rg, "RG") != 0){
            printf("opcao invalida\n");
            continue;
        }

        printf("Digite a idade do RG?[formato:DD/MM/YYYY]");
        fgets(dataRg, sizeof(dataRg), stdin);
        dataRg[strcspn(dataRg, "\n")] = '\0';
        if(dataRg[2]!='/' || dataRg[5]!='/'){
            printf("data invalida\n");
            continue;
        }
        printf("Voce possui Passagem?[se sim digite: Passagem/se nao digite: Nao possui]");
        fgets(Passagem, sizeof(Passagem), stdin);
        Passagem[strcspn(Passagem, "\n")] = '\0';
        if (strcmp(Passagem, "Nao possui") == 0){
            printf("A recepcao e nessa direcao\n");
            continue;
        }
          else if (strcmp(Passagem, "Passagem") != 0){
            printf("opcao invalida\n");
            continue;
        }

        printf("Digite a idade da Passagem[formato:DD/MM/YYYY]:");
        fgets(dataPass, sizeof(dataPass), stdin);
        dataPass[strcspn(dataPass, "\n")] = '\0';
        if(dataPass[2] != '/' || dataPass[5] != '/'){
            printf("data invalida\n");
            continue;
        }
        else if(strcmp(dataRg, dataPass) != 0){
            printf("190\n");
            continue;
        }
        printf("Digite a cadeira do passageiro[formato:A12]:");
        fgets(cadeiraPass, sizeof(cadeiraPass), stdin);
        cadeiraPass[strcspn(cadeiraPass, "\n")] = '\0';
        if( !isalpha(cadeiraPass[0]) || isalpha(cadeiraPass[1]) || isalpha(cadeiraPass[2])){
            printf("Formato invalido\n");
            continue;
        }
        printf("seu assento e %s, tenha um otimo dia\n", cadeiraPass);
    }
    return 0;
}

File: aeroporto.java
import java.util.Scanner;
public class aeroporto {
    
    public static void main(String[] args) {
        System.out.println("Bem vindo ao Aeroporto");
        Scanner sc = new Scanner(System.in);
        System.out.print("digite a quantidade de passageiros:");
        int qntPass = sc.nextInt();
        sc.nextLine();
        System.out.printf("total de passageiro: %d\n", qntPass);
        for(int i = 0; i < qntPass; i++){
            System.out.printf("Passageiro %d\n", i+1);
            System.out.print("Voce possui RG?[se sim digite: RG/se nao digite: Nao possui]");
            String rg = sc.nextLine();
            if(rg.toUpperCase().equals("NAO POSSUI")){
                System.out.println("A saida e nessa direção");
                continue;
            }
            else if(rg.toUpperCase().equals("RG") == false){
                System.out.println("Opção inválida");
                continue;
            }
            System.out.print("Digite a idade do RG?[formato:DD/MM/YYYY]");
            String dataRg = sc.nextLine();
            if(dataRg.charAt(2) != '/' || dataRg.charAt(5) != '/'){
                System.out.println("Data inválida");
                continue;
            }
            System.out.print("Você possui Passagem?[se sim digite: passagem/se nao digite: Nao possui]");
            String pass = sc.nextLine();
            if(pass.toLowerCase().equals("nao possui")){
                System.out.println("\"A recepcao e nessa direcao");
                continue;
            }
            else if(pass.toUpperCase().equals("PASSAGEM") == false){
                System.out.println("Opção inválida");
                continue;
            }
            System.out.print("Digite a idade da Passagem[formato:DD/MM/YYYY]:");
            String dataPass = sc.nextLine();
            if(dataPass.charAt(2) != '/' || dataPass.charAt(5) != '/'){
                System.out.println("Data inválida");
                continue;
            }
            else if(dataPass.equals(dataRg) == false){
                System.out.println("190");
                continue;
            }
            System.out.print("Digite a cadeira[formato: A12]:");
            String cad = sc.nextLine();
            if(!Character.isLetter(cad.charAt(0)) || Character.isLetter(cad.charAt(1)) || Character.isLetter(cad.charAt(2))){
                System.out.println("Cadeira inválida");
                continue;
            }
            else{
                System.out.printf("seu assento e %s, tenha um otimo dia\n", cad);
            }
        }
        sc.close();
    }
    
}
